google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java

```diff
@@ -83,6 +83,30 @@
    */
   Timestamp writeAtLeastOnce(Iterable<Mutation> mutations) throws SpannerException;
 
+  /**
+   * Response for write methods that return both a commit {@link Timestamp} and {@link CommitStats}.
+   */
+  interface WriteResponse {
+    /** The commit timestamp of the transaction. */
+    Timestamp getCommitTimestamp();
+
+    /** The commit statistics of the transaction. */
+    CommitStats getCommitStats();
+  }
+
+  /**
+   * Same as {@link #write(Iterable)}, but requests the backend to return both a commit timestamp
+   * and {@link CommitStats}.
+   */
+  WriteResponse writeWithCommitStats(Iterable<Mutation> mutations) throws SpannerException;
```

Comment:
Can't we add a breaking change to the existing `write` methods in order to avoid the overload here?

Reply:
That is possible, but it will be a compile time breaking change which will require users to change existing code if they use the commit timestamp that was returned by the `write` methods. I've implemented it that way now, and it does seem like a better solution to me as well, assuming that we do accept this breaking change.
The advantage of this solution is that we can introduce a `TransactionOption` option in the same way as that we already have `ReadOption` and `QueryOption`, and use that for all transaction types that need to support `CommitStats`. This also adds additional possibilities for future additions to read/write transactions and/or commit requests.

WDYT?

Reply:
Hey @olavloite, we had an internal discussion and decided not change the existing methods (sorry about that). Instead, we decided to create two new methods, `writeWithOptions` and `writeAtLeastOnceWithOptions` that would have the different return type, as well as accept `WriteOption`s.

Since this modification will be required here, and for another feature that we are developing, we decided to put the API change in a separate PR. This way we can merge that in first and use it both here and for the other feature. You can take a look at the PR in https://github.com/googleapis/java-spanner/pull/531.